pkgs/by-name/fl/flatpak/package.nix

```diff
@@ -89,163 +94,137 @@ stdenv.mkDerivation (finalAttrs: {
+      wayland
+      wayland-protocols
+      wayland-scanner
+      xorg.libXau
+      zstd
+    ]
+    ++ lib.optional withGtkDoc gtk-doc
+    ++ lib.optional withSystemd systemd;
 
   # Required by flatpak.pc
   propagatedBuildInputs = [
     glib
     ostree
   ];
 
-  nativeCheckInputs = [ valgrind ];
-
-  # TODO: some issues with temporary files
-  doCheck = false;
-  strictDeps = true;
-
-  NIX_LDFLAGS = "-lpthread";
-
-  enableParallelBuilding = true;
-
-  configureFlags = [
-    "--with-curl"
-    "--with-system-bubblewrap=${lib.getExe bubblewrap}"
-    "--with-system-dbus-proxy=${lib.getExe xdg-dbus-proxy}"
-    "--with-dbus-config-dir=${placeholder "out"}/share/dbus-1/system.d"
-    "--with-profile-dir=${placeholder "out"}/etc/profile.d"
-    "--localstatedir=/var"
-    "--sysconfdir=/etc"
-    "--enable-gtk-doc=${if withGtkDoc then "yes" else "no"}"
-    "--enable-installed-tests"
-    "--enable-selinux-module=no"
+  mesonFlags = [
+    (lib.mesonBool "auto_sideloading" withAutoSideloading)
+    (lib.mesonBool "installed_tests" finalAttrs.finalPackage.doCheck)
+    (lib.mesonBool "tests" finalAttrs.finalPackage.doCheck)
+    (lib.mesonEnable "selinux_module" withSELinuxModule)
+    (lib.mesonOption "dbus_config_dir" (placeholder "out" + "/share/dbus-1/system.d"))
+    (lib.mesonOption "profile_dir" (placeholder "out" + "/etc/profile.d"))
+    (lib.mesonOption "system_bubblewrap" (lib.getExe bubblewrap))
+    (lib.mesonOption "system_dbus_proxy" (lib.getExe xdg-dbus-proxy))
+    (lib.mesonOption "system_fusermount" "/run/wrappers/bin/fusermount3")
+    (lib.mesonOption "system_install_dir" "/var/lib/flatpak")
   ];
 
-  makeFlags = [
-    "installed_testdir=${placeholder "installedTests"}/libexec/installed-tests/flatpak"
-    "installed_test_metadir=${placeholder "installedTests"}/share/installed-tests/flatpak"
+  nativeCheckInputs = [
+    polkit
+    socat
+    valgrind
   ];
 
-  postPatch =
-    let
-      vsc-py = python3.pythonOnBuildForHost.withPackages (pp: [ pp.pyparsing ]);
-    in
-    ''
-      patchShebangs buildutil
-      patchShebangs tests
-      PATH=${lib.makeBinPath [ vsc-py ]}:$PATH patchShebangs --build subprojects/variant-schema-compiler/variant-schema-compiler
+  # TODO: Many issues with temporary files, FHS environments, timeouts, and our current patches
+  doCheck = false;
 
-      substituteInPlace configure.ac \
-        --replace-fail '$BWRAP --' ${
-          lib.escapeShellArg (stdenv.hostPlatform.emulator buildPackages + " $BWRAP --")
-        } \
-        --replace-fail '$DBUS_PROXY --' ${
-          lib.escapeShellArg (stdenv.hostPlatform.emulator buildPackages + " $DBUS_PROXY --")
-        }
-    '';
+  separateDebugInfo = true;
 
   passthru = {
     icon-validator-patch = substituteAll {
       src = ./fix-icon-validation.patch;
       inherit (builtins) storeDir;
     };
 
-    updateScript = nix-update-script { };
-
     tests = {
-      cross = pkgsCross.aarch64-multiplatform.flatpak;
```

Comment:
@getchoo curious why the cross compilation test was dropped? if we can keep that test in whatever way better fits your packaging idioms let me know & i'd be happy to PR that :)

Reply:
Just a merge conflict mistake. I'm fine with keeping it as I tend to use `pkgsCross` a lot as well